deepspeed/runtime/engine.py

```diff
@@ -2303,7 +2303,10 @@ def _get_gradients_for_reduction(self):
                 expert_grads[key] = []
 
         for param_name, param in self.module.named_parameters():
-            if param.grad is None and param.requires_grad:
+            if not param.requires_grad:
+                continue
+
+            if param.grad is None:
```

Comment:
@awan-10 The CI failures is due to https://github.com/microsoft/DeepSpeed/pull/3789.

After the change in that PR, `param.grad` won't be set when `param.grad` is None, and `param.grad.data` will trigger the error.

Reply:
Thank you for the fix @sxjscience.

@jomayeri and @jeffra - please review this part, resolve this conversation, and approve the PR, so we can merge it.

Reply:
Looks good.